diff --git a/core/lib/oauth/client.js b/core/lib/oauth/client.js
index c2f0d40e6bbbdc2eaebddb4c19b8358c22837ef7..b2d6489f85d8bf8c1a07974242d71327d83604be 100644
--- a/core/lib/oauth/client.js
+++ b/core/lib/oauth/client.js
@@ -7,8 +7,12 @@ exports.openidClient = openidClient;
 
 var _openidClient = require("openid-client");
 
+var ProxyAgent = require("proxy-agent");
+
 async function openidClient(options) {
   const provider = options.provider;
+  let httpOptions = { ...provider.httpOptions };
+  httpOptions.agent = new ProxyAgent();
   if (provider.httpOptions) _openidClient.custom.setHttpOptionsDefaults(provider.httpOptions);
   let issuer;
 
